### ISSUE TITLE
Update MGLVectorTileSource for Mapbox Streets v8 compatibility

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-gl-native#11867]
MGLVectorTileSource currently hard-codes compatibility with Mapbox Streets source v6 and v7, particularly for the built-in label localization functionality. When v8 comes out, we’ll need to update MGLVectorTileSource to handle this version as well. In addition to rote version bumping, we’ll also have to account for the fact that `name_*` fields in v8 may be unset if the value would be identical to `name` or `name_en` (need to confirm the details). Finally, the VoiceOver accessibility code may need updates to recognize any new or renamed fields in v8.

https://github.com/mapbox/mapbox-gl-native/blob/d1dff1d6e782bc6e9d7a7df27e566bf1f1cf862b/platform/darwin/src/MGLVectorTileSource.mm#L129-L136
https://github.com/mapbox/mapbox-gl-native/blob/d1dff1d6e782bc6e9d7a7df27e566bf1f1cf862b/platform/darwin/src/MGLVectorTileSource.mm#L78-L94
https://github.com/mapbox/mapbox-gl-native/blob/d1dff1d6e782bc6e9d7a7df27e566bf1f1cf862b/platform/ios/src/MGLMapAccessibilityElement.mm#L80-L93

* [x] Detect v8 sources by URL: #13481
* [x] Enumerate v8 `name_*` languages: #13481
* [x] Coalesce `name_*` key paths with `name` or `name_en` in expressions when localizing labels: #12387
* [ ] Avoid making expressions too complex when relocalizing: https://github.com/mapbox/mapbox-gl-native/pull/12387#discussion_r204784160
* [x] (While we’re at it) Remove `-localizedKeysByKeyForPreferredLanguage:`, which is unused: #11907
* [x] Update MGLMapAccessibilityElement.mm with v8 field names: #13481
* [ ] Consider defining an `MGLVectorTileSource.mapboxStreetsURL` class property

This work is not blocked by v8 launching; in fact, we should have it in place and released ahead of the launch, so that styles using v8 can take advantage of label localization.

/cc @fabian-guerra @friedbunny @nickidlugash

**Comment by 1ec5** [comment]
> When v8 comes out, we’ll need to update MGLVectorTileSource to handle this version as well. In addition to rote version bumping, we’ll also have to account for the fact that `name_*` fields in v8 may be unset if the value would be identical to `name` or `name_en` (need to confirm the details).

#12387 uses a coalescing expression to fall back from `name_*` to `name`. I think that’ll get us into a good spot for basic Streets source v8 usage. However, it’s also possible that a future style would take advantage of expressions to implement bilingual labels like `London (Londres)`. In that case, we’ll have to be a bit less aggressive in localizing conditional expressions and string-concatenating expressions. But we can worry about that in a second pass, after addressing basic usage.

/cc @lloydsheng

**Comment by 1ec5** [comment]
#12387 landed. https://github.com/mapbox/mapbox-gl-native/pull/12387#discussion_r204784160 describes how to more gracefully handle the case where a layer gets relocalized, which could be important for performance. The remaining work for this issue is tracked in the to-do list above.

**Comment by 1ec5** [comment]
Here’s [a gist](https://gist.github.com/1ec5/9ccb9d34db3c723ff7240a8a79f05e80) where I had made some baby steps. The main task was allowing for multiple languages to be “preferred” on the client side. Hopefully these changes hook into the changes made in mapbox/mapbox-gl-native#12164.

**Comment by friedbunny** [comment]
> Update MGLMapAccessibilityElement.mm with v8 field names

Looking at the [Streets-v8 changelog](https://www.mapbox.com/vector-tiles/mapbox-streets-v8/#changelog), attributes have been moved around and added, but it doesn’t appear to me that anything will break.

- `elevation_m` & `elevation_ft` are unchanged.
- `road_label` has been merged into `road`, but keeps the same attribute names.

**Comment by 1ec5** [comment]
#13481 and #13525 have implemented the bulk of the work for this issue. The following task is unfinished, but it seems lower in priority:

> Avoid making expressions too complex when relocalizing: https://github.com/mapbox/mapbox-gl-native/pull/12387#discussion_r204784160
